File: ObjectiveC/FlexGrid101/FlexGrid101/SelectionModesController.m
//
//  SelectionModesController.m
//  FlexGrid101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//

#import "SelectionModesController.h"
#import "CustomerData.h"
#import "XuniFlexGridDynamicKit/XuniFlexGridDynamicKit.h"

@interface SelectionModesController ()

@property (weak, nonatomic) IBOutlet FlexGrid           *flex;
@property (weak, nonatomic) IBOutlet UISegmentedControl *modeSwitch;
@property (weak, nonatomic) IBOutlet UIBarButtonItem    *selected;
@property (weak, nonatomic) IBOutlet UIBarButtonItem    *selectedCount;

@end

@implementation SelectionModesController
- (IBAction)modeSwitched:(id)sender {
    NSInteger row = self.modeSwitch.selectedSegmentIndex;

    if (row == 0) {
        self.flex.selectionMode = GridSelectionModeNone;
    } else if (row == 1) {
        self.flex.selectionMode = GridSelectionModeCell;
    } else if (row == 2) {
        self.flex.selectionMode = GridSelectionModeCellRange;
    } else if (row == 3) {
        self.flex.selectionMode = GridSelectionModeRow;
    } else if (row == 4) {
        self.flex.selectionMode = GridSelectionModeRowRange;
    }
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self.flex.isReadOnly       = true;
    self.flex.columnHeaderFont = [UIFont boldSystemFontOfSize:self.flex.columnHeaderFont.pointSize];
    self.flex.selectionMode    = GridSelectionModeNone;
    self.flex.itemsSource      = [CustomerData getCustomerData:100];
    
    [self.flex.flexGridSelectionChanged addHandler:^(XuniEventContainer<GridCellRangeEventArgs *> *eventContainer) {
        int selected = self.flex.selection.columnSpan * self.flex.selection.rowSpan;
        
        self.selectedCount.title = [NSString stringWithFormat:@"%i selected", selected];
    } forObject:self];
}

@end

File: Swift/FlexGrid101/FlexGrid101/SelectionModesController.swift
//
//  SelectionModesController.h
//  FlexGrid101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//
import UIKit
import XuniCoreDynamicKit
import XuniFlexGridDynamicKit

class SelectionModesController: UIViewController {

    @IBAction func modeSwitched(sender: AnyObject) {
        let row: Int = self.modeSwitch.selectedSegmentIndex
        if row == 0 {
            self.flex.selectionMode = GridSelectionMode.None
        }
        else if row == 1 {
            self.flex.selectionMode = GridSelectionMode.Cell
        }
        else if row == 2 {
            self.flex.selectionMode = GridSelectionMode.CellRange
        }
        else if row == 3 {
            self.flex.selectionMode = GridSelectionMode.Row
        }
        else if row == 4 {
            self.flex.selectionMode = GridSelectionMode.RowRange
        }

    }

    override func viewDidLoad() {
        super.viewDidLoad()
        self.flex.isReadOnly = true
        self.flex.columnHeaderFont = UIFont.boldSystemFontOfSize(self.flex.columnHeaderFont.pointSize)
        self.flex.selectionMode = GridSelectionMode.None
        self.flex.itemsSource = NSMutableArray(array:CustomerData.getCustomerData(100))
        self.flex.flexGridSelectionChanged.addHandler({(eventContainer: XuniEventContainer!) -> Void in
            let selected = self.flex.selection.columnSpan * self.flex.selection.rowSpan
            self.selectedCount.title = "\(selected) selected"
        }, forObject: self)
    }

    @IBOutlet weak var flex: FlexGrid!
    @IBOutlet weak var modeSwitch: UISegmentedControl!
    @IBOutlet weak var selected: UIBarButtonItem!
    @IBOutlet weak var selectedCount: UIBarButtonItem!
}
//
//  SelectionModesController.m
//  FlexGrid101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//